scripts/gen_syscalls.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import re
+import argparse
+import os
+
+
+api_regex = re.compile(r'''
+__(syscall|syscall_inline)\s+   # __syscall or __syscall_inline
+static\s+inline\s+              # All prototypes are static inline functions
```

Comment:
The __syscall looks like a macro that's going to expand to a macro, and GCC allows attributes to be before or after the static, or (I think) after the argument list.  This might be stricter than many coders are going to expect.

Alternatively you can write a not-too-complicated regex to handle exactly one level of nesting, which would unbreak the common case of a function pointer.

Reply:
yeah for the moment it has to be the very first thing.
planning on addressing this when I beef this up to handle DWARF info